src/Components/MovieListPage/MovieListPage.js

```diff
@@ -0,0 +1,124 @@
+import React, {useState, useEffect, useCallback} from "react";
+import styles from './MovieListPage.module.scss';
+import GenreSelector from '../GenreSelector/GenreSelector';
+import SortControl from '../SortControl/SortControl';
+import MoviesFound from '../MoviesFound/MoviesFound';
+import MovieTile from '../MovieTile/MovieTile';
+import Logo from '../Logo/Logo';
+import Header from '../Header/Header';
+import MovieDetails from "../MovieDetails/MovieDetails";
+import SwitchComponents from "../SwitchComponents/SwitchComponents";
+import {GenreListArray, SortControlArray} from '../../data';
+
+const MovieListPage = () => {
+    const [activeComponent, setActiveComponent] = useState("header");
+    const [selectedMovie, setSelectedMovie] = useState(null);
+    const [searchQuery, setSearchQuery] = useState('');
+    const [sortCriterion, setSortCriterion] = useState('');
+    const [activeGenre, setActiveGenre] = useState('all');
+    const [movieList, setMovieList] = useState([]);
+    const [movieListLength, setMovieListLength] = useState(movieList.length);
+    const [isLoading, setIsLoading] = useState(false);
+    const [controller, setController] = useState(new AbortController());
+    let queryParams = new URLSearchParams();
```

Comment:
You don't reassign that variable, so you can use const as for the other variables